src/Workspaces/Core/Portable/Utilities/BKTree.Builder.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    internal partial class BKTree
+    {
+        private class Builder
+        {
+            private const int CompactEdgeAllocationSize = 4;
+
+            private readonly char[][] _values;
+
+            // Note: while building a BKTree we have to store children with parents, keyed by the
+            // edit distance between the two.  Naive implementations might store a list or dictionary
+            // of children along with each node.  However, this would be very inefficient and would
+            // put an enormous amount of memory pressure on the system.
+            //
+            // Imperical data for a nice large assembly like mscorlib gives us the following 
```

Comment:
This is in Utilities, so you should probably add more context here, that you added all *names* in mscorlib and got this distribution.